GliaWidgets/SecureConversations/SecureConversations.ChatWithTranscriptModel.swift

```diff
@@ -429,26 +437,116 @@ extension SecureConversations {
             action?(.sendButtonHidden(!isValid))
             return isValid
         }
+    }
+}
 
-        private func sendMessage() {
-            guard validateMessage() else { return }
-
-           _ = environment.sendSecureMessage(
-                messageText,
-                fileUploadListModel.attachment,
-                environment.queueIds
-           ) { [weak self] result in
-               switch result {
-               case .success:
-                   self?.messageText = ""
-                   self?.fileUploadListModel.removeSucceededUploads()
-                   // TODO: MOB-1738
-               case let .failure(error):
-                   // TODO: MOB-1874
-                   break
-               }
+// MARK: Message management
+extension SecureConversations.TranscriptModel {
+    private func sendMessage() {
+        guard validateMessage() else { return }
+
+        let uploads = fileUploadListModel.succeededUploads
+        let localFiles = uploads.map(\.localFile)
+
+        let outgoingMessage = OutgoingMessage(
+            content: messageText,
+            files: localFiles
+        )
+        appendItem(.init(kind: .outgoingMessage(outgoingMessage)), to: messagesSection, animated: true)
+
+       _ = environment.sendSecureMessage(
+            messageText,
+            fileUploadListModel.attachment,
+            environment.queueIds
+       ) { [weak self] result in
+           guard let self = self else { return }
+           switch result {
+           case let .success(message):
+               self.replace(
+                outgoingMessage,
+                uploads: uploads,
+                with: message,
+                in: self.messagesSection
+               )
+           case let .failure(error):
+               // TODO: MOB-1874
+               break
            }
-        }
+       }
+
+        // Clear inputs, thus disabling them
+        // and preventing the message to be sent again.
+        messageText = ""
+        fileUploadListModel.removeSucceededUploads()
+        action?(.scrollToBottom(animated: true))
+    }
+
+    func replace(
```

Comment:
NIT: Same, since we are stuck with having to bring over things from chat screen, at least in my opinion, they could be made slightly easier to work with if we separate this into a couple of smaller scopes to make it easier to read and modify in the future.

Reply:
I think maybe it makes sense to have it in one place, instead of copying overt here. However I would like that to be done apart from this PR to avoid distraction from the feature development. I will create a separate ticket for this.